src/grammar.pegjs

```diff
@@ -3,12 +3,12 @@
   const {
     Assignment,
     BinaryOp,
-    BooleanLiteral,
     Catch,
     ClassDeclaration,
     Closure,
     ConstructorDeclaration,
     FeatureCall,
+    FieldDeclaration,
```

Comment:
Regarding FieldDeclaration, I created it because variable declaration seemed to go on a complete different way. They compile to different things and It was annoying that it had a "different" meaning depending on the context. It looked like it was just a coincidence that the words being used are the same and, eventually that may change for fields (for example, if properties are added to the language). It felt like it belong. Like the Block.

About the variable... I have wonder about that myself. On one hand, it may come in handy some structure to represent any extra data besides the name (perhaps for the linker, or the validator), similar to what happens with Parameter (we need the Node so we can tell if it's a vararg). On the other hand, you may say the same for many other "strings" are left alone (the feature access name is one example), so why can't this one too?

To be honest, I just left that like I found it on Xtext, but we could remove it and see if we ever need it.

On a similar line of though, more than once I considered that the Self Node could be just a regular variable and let the linker handle any extra work.